src/mocks/data.ts

```diff
@@ -363,17 +363,17 @@ export const mockUserToUpdate = {
   schoolOther: undefined,
 };
 
-export const buildMockStudent = <T extends number>(
-  id: T extends typeof mockUser.id ? `Student ID cannot be the same as the mockUser: ${typeof mockUser.id}` : T,
-) => {
-  if (id === mockUser.id) throw Error("A mock student cannot have the same ID as the mockUser");
+export const buildMockStudent = <T>(id: T) => {
```

Comment:
What do we need generics for? It looks like `id` should always be a number, right? Removing the generics would make this function simpler.